examples/distributed-key-value-store.rs

```diff
@@ -157,35 +154,23 @@ async fn main() -> Result<(), Box<dyn Error>> {
     };
 
     // Read full lines from stdin
-    let mut stdin = io::BufReader::new(io::stdin()).lines();
+    let mut stdin = io::BufReader::new(io::stdin()).lines().fuse();
 
     // Listen on all interfaces and whatever port the OS assigns.
     swarm.listen_on("/ip4/0.0.0.0/tcp/0".parse()?)?;
 
     // Kick it off.
-    task::block_on(future::poll_fn(move |cx: &mut Context<'_>| {
-        loop {
-            match stdin.try_poll_next_unpin(cx)? {
-                Poll::Ready(Some(line)) => {
-                    handle_input_line(&mut swarm.behaviour_mut().kademlia, line)
-                }
-                Poll::Ready(None) => panic!("Stdin closed"),
-                Poll::Pending => break,
-            }
-        }
-        loop {
-            match swarm.poll_next_unpin(cx) {
-                Poll::Ready(Some(event)) => {
-                    if let SwarmEvent::NewListenAddr { address, .. } = event {
-                        println!("Listening on {:?}", address);
-                    }
-                }
-                Poll::Ready(None) => return Poll::Ready(Ok(())),
-                Poll::Pending => break,
+    loop {
+        select! {
+            line = stdin.next() => handle_input_line(&mut swarm.behaviour_mut().kademlia, line.expect("Stdin not to close")?),
```

Comment:
For the behaviour, we use `select_next_some` to unwrap the `Option`. I think that should work here as well?

It won't panic in that case because `select` is specifically designed to work well with `FusedFuture`: https://docs.rs/futures/0.3.18/futures/stream/trait.StreamExt.html#method.select_next_some

Reply:
I have used `select_next_some` for all places where we use `stdin`, except in `chat-tokio.rs`, because `tokio` does not have `select_next_some`, like in `async_std`.